codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/http/HttpBindingGenerator.kt

```diff
@@ -0,0 +1,383 @@
+import software.amazon.smithy.rust.codegen.rustlang.withBlock
+import software.amazon.smithy.rust.codegen.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.makeOptional
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpBindingDescriptor
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpLocation
+import software.amazon.smithy.rust.codegen.smithy.protocols.Protocol
+import software.amazon.smithy.rust.codegen.smithy.protocols.parse.EventStreamUnmarshallerGenerator
+import software.amazon.smithy.rust.codegen.smithy.rustType
+import software.amazon.smithy.rust.codegen.util.UNREACHABLE
+import software.amazon.smithy.rust.codegen.util.dq
+import software.amazon.smithy.rust.codegen.util.hasTrait
+import software.amazon.smithy.rust.codegen.util.isPrimitive
+import software.amazon.smithy.rust.codegen.util.isStreaming
+import software.amazon.smithy.rust.codegen.util.toSnakeCase
+
+/**
+ * The type of HTTP message from which we are deserializing the HTTP-bound data.
+ * We are either:
+ *     - deserializing data from an HTTP request (we are a server); or
+ *     - deserializing data from an HTTP response (we are a client).
+ */
+enum class HttpMessageType {
+    REQUEST, RESPONSE
+}
+
+/**
+ * This class generates Rust functions that deserialize data from an HTTP message.
+ * They are useful for *both*:
+ *     - servers (that deserialize HTTP requests); and
+ *     - clients (that deserialize HTTP responses)
+ * because the deserialization logic is entirely similar. In the cases where they _slightly_ differ,
+ * the [HttpMessageType] enum is used to distinguish.
+ *
+ * For deserialization logic that is wholly different among clients and servers, use the classes:
+ *     - [ServerRequestBindingGenerator] from the `codegen-server` project for servers; and
+ *     - [ResponseBindingGenerator] from this project for clients
+ * instead.
+ */
+class HttpBindingGenerator(
+    private val protocol: Protocol,
+    codegenContext: CodegenContext,
+    private val operationShape: OperationShape
+) {
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val symbolProvider = codegenContext.symbolProvider
+    private val mode = codegenContext.mode
+    private val model = codegenContext.model
+    private val service = codegenContext.serviceShape
+    private val index = HttpBindingIndex.of(model)
+    private val headerUtil = CargoDependency.SmithyHttp(runtimeConfig).asType().member("header")
+    private val defaultTimestampFormat = TimestampFormatTrait.Format.EPOCH_SECONDS
+    private val dateTime = RuntimeType.DateTime(runtimeConfig).toSymbol().rustType()
+    private val httpSerdeModule = RustModule.private("http_serde")
+
+    /**
+     * Generate a function to deserialize [binding] from HTTP headers
+     *
+     * The name of the resulting function is returned as a String
```

Comment:
```suggestion
     * Generates a function to deserialize [binding] from HTTP headers.
     *
     * The name of the resulting function is returned as a String.
```

Reply:
I moved `ResponseBindingGenerator.kt` to `HttpBindingGenerator.kt`; all the existing comments use the imperative. I'll add the full stops though.